pytools/graph.py

```diff
@@ -45,13 +46,37 @@
     Any type.
 """
 
-from typing import (TypeVar, Mapping, Iterable, List, Optional, Any, Callable,
-                    Set, MutableSet, Dict, Iterator, Tuple)
+from typing import (Collection,  TypeVar, Mapping, Iterable, List, Optional, Any,
+                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet)
 
 
 T = TypeVar("T")
 
 
+# {{{ reverse_graph
+
+def reverse_graph(graph: Mapping[T, Collection[T]]) -> Dict[T, FrozenSet[T]]:
+    """
+    Reverses a graph.
+
+    :param graph: A :class:`dict` representation of a directed graph, mapping each
+        node to other nodes to which it is connected by edges.
+    :returns: A :class:`dict` representing *graph* with edges reversed.
+    """
+    result: Dict[T, Set[T]] = {}
+
+    for node_key, edges in graph.items():
```

Comment:
```suggestion
    for node_key, successors in graph.items():
```
"edges" makes it sounds like the values are a tuple pair.